lib/DDG/Spice/Currency.pm

```diff
@@ -113,13 +113,28 @@ sub checkCurrencyCode {
             $to = $from eq 'usd' ? 'eur' : 'usd';
         }
     }
-    
+
     return $normalized_number, $from, $to;
 }
 
 handle query_lc => sub {
     if(/$guard/) {
-        my ($amount, $from, $alt_amount, $to) = ($1, $2, $3, $4 || '');
+        my ($amount, $cardinal, $from, $alt_amount, $to) = ($1, $2 || '', $3, $4, $5 || '');
+
+        if ($amount =~ m/k$/) {
+            $amount =~ s/(\,|k)//g;
```

Comment:
You should let the methods in `DDG::SpiceRole::NumberStyler` process the number for you instead of removing the commas yourself. See what happens with the query "convert 4.000.000k euro to usd":

![screen shot 2015-04-08 at 1 21 15 am](https://cloud.githubusercontent.com/assets/81969/7039284/a08acd42-dd8d-11e4-8441-64f5305ae82c.png)
It tries to remove commas, but the number doesn't have any because it's in the European format.